src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLY_COEFFICIENT = 2;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (threshold == size) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> node = table[index];
+        if (node == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> foundNode = findNode(node, key);
+            if (foundNode != null) {
+                foundNode.value = value;
+            } else {
+                Node<K, V> lastNode = getLastNode(node);
+                lastNode.next = new Node(key, value, null);
+                size++;
+            }
```

Comment:
This algorithm is not very optimal, because you need to go through the entire list of nodes twice. Can we do it in 1 pass?

Reply:
Changed places the conditions